### ISSUE TITLE
Torch on the head

### ISSUE THREAD

**Issue by MaGeBZz** [GTNewHorizons/GT-New-Horizons-Modpack#17444]
### Your GTNH Discord Username

marvinhhmagebzz

### Your Pack Version

2.7 Beta 1

### Your Server

SP

### Java Version

Java 17

### Type of Server

Single Player

### Your Expectation

That shouldn't happen I guess lol

![image](https://github.com/user-attachments/assets/82e7697d-3048-43f1-8834-f8e8001a065f)

### The Reality

You can place a torch on your head in the tab (look screenshot) and it will still emit light even if it's there

### Your Proposal

Fix that the torch can't be on the head slot haha

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [X] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

**Comment by leumasme** [comment]
The having blocks on your head part is intended by some mod as a cosmetic thing, pretty sure (should be visible on player). Having all visible slots of the player activate dynamic lights is expected/makes sense and is pretty neat. Maybe for blocks that won't properly be rendered on head like torches (what's going on there lol, it's floating around behind the player?) it could be changed, but ex. for blocks that should(?) render fine glowstone block this seems fine imo.

**Comment by NotAPenguin0** [comment]
Agree, I think this isn't really a bug

**Comment by chochem** [comment]
this really sounds like a bug, weird. what mod 'intends' stuff like that?

**Comment by MaGeBZz** [comment]
@chochem would say so too
I mean it's not a huge bug or so but still it's weird although it's a good alternative cause you don't have a real off-hand like in newer minecraft versions to hold a torch the whole time in it while mine things for example

**Comment by leumasme** [comment]
I don't know which mod adds it but placing blocks on your head is a common feature even on modern servers (/head command), since the game already has functionality to render a block on the player head slot so many just let the player use it as a cosmetic thing; so I assume it's intended by some mod, especially since it's only the head slot.
If there's intent to remove it then maybe we could think about getting some type of earlygame-ish helmet that emits dynamic light? I think it's really neat.

**Comment by MaGeBZz** [comment]
@leumasme yeah I guess it's just a bit weird with the torch but if it's on me it doesn't have to be fixed but wanna mention it at least and maybe getting discussed
But yeah only head slot
But yeah with dynamic lights now (from angelica I guess?) I think a helmet with dynamic light like this bug would be really neat
Maybe something like torch + leather strap kinda (really just a small idea rn lol)

**Comment by chochem** [comment]
ok I tried and this doesnt actually seem to be a thing. no torch in head slot.

**Comment by chochem** [comment]
ok so it only works when you put it on in a strange way and the rendering is all wrong. there are certainly bugs here. even if its a botched feature underneath.

**Comment by Caedis** [comment]
The dynamic lights working with it is intended since I liked the "feature" from OF

**Comment by chochem** [comment]
TLDR: if its meant to be a feature then its bugged in the regular GUI and clearly the visuals are bugged to:
![image](https://github.com/user-attachments/assets/e3e4cce5-1df9-47d3-b237-4ec08eae468a)

**Comment by Caedis** [comment]
@chochem it is 100% intended by TiCon. It has also existed for over 11 years.

https://github.com/GTNewHorizons/TinkersConstruct/blob/master/src/main/java/tconstruct/armor/inventory/SlotArmorCopy.java#L52
```
/**
     * Check if the stack is a valid item for this slot. Always true beside for the armor slots.
     */
    @Override
    public boolean isItemValid(ItemStack par1ItemStack) {
        Item item = (par1ItemStack == null ? null : par1ItemStack.getItem());
        boolean isValidArmor = false;
        if (item instanceof ItemArmor) {
            isValidArmor = (((ItemArmor) item).armorType == armorType);
        }
        return item != null && (isValidArmor || (item instanceof ItemBlock && armorType == 0));
    }
```

armorType 0 is a helm (vanilla)
![image](https://github.com/user-attachments/assets/0f8968a7-3bbc-4495-b607-724caf266cbc)

**Comment by chochem** [comment]
still very buggy lol. we have been over this 3 times now.
